server/api/controllers/exam-bank-controller.ts

```diff
@@ -0,0 +1,110 @@
+import { Dirent, readdirSync } from "fs";
+import { Exam } from "../../types/exam-bank";
+import { Logger } from "../../util/logger";
+import { ReadWriteController } from "./read-write-controller";
+
+export class ExamBankController {
+  static logger = new Logger("Exam Bank Controller");
+
+  /**
+   * Re-creates the exam list JSON file based on the current 
+   * contents of the exams directory 
+   */
+  static rewriteFile(): void {
+    const url = "_hidden/exams-list";
+    ReadWriteController.overwriteJSONDataPath(
+      url,
+      (statusCode: number) => {
+        switch (statusCode) {
+          case 400:
+            this.logger.error(`Bad request made`);
+            break;
+          case 403:
+            this.logger.error(`Forbidden`);
+            break;
+          case 404:
+            this.logger.error(`${url} not found`);
+            break;
+          case 200:
```

Comment:
is `case 200:` supposed to be empty?

Reply:
yes - I just changed the structure to be clearer.